Replace und with en

garemoko opened the pull request:
Because all the verbs in _downConvertMap are English.

garemoko commented:
@brianjmiller this is ready to review (actually was ready 22 days ago but was waiting for tests to complete and forgot to come back and check)

brianjmiller commented on src/Verb.js:
```diff
@@ -97,7 +97,7 @@ TinCan client library
                 this.id = cfg;
                 this.display = {
-                    und: this.id
+                    en: this.id
```
This is what they want to use for the 'id' and we can't know whether it is English as it is an IRI (or the special 0.9 case where we do know, see additional comments). Now it could be argued that in that case we shouldn't be setting a 'display' property at all.

garemoko replied:
Gotcha. +1 to not using display property at all here.

brianjmiller commented on src/Verb.js:
```diff
@@ -120,7 +120,7 @@ TinCan client library
                 if (this.display === null && typeof _downConvertMap[this.id] !== "undefined") {
                     this.display = {
-                        "und": _downConvertMap[this.id]
+                        "en": _downConvertMap[this.id]
```
While we know they are English, I'm not sure this is worth the backwards compatible change. Any client should be accepting "und", so I'd expect this to only negatively effect foreign clients where they don't want to see an English word in a place where they aren't expecting English, except the issue is with 0.9 use cases only and the spec didn't allow for non-English use in that case so they only had one option, that of the English.

Ultimately 0.9 support gets dropped completely in the 1.x roadmap so I'm also inclined to leave this as is until that point and it becomes a non-issue.

garemoko replied:
I'm not sure this does go away with 0.9 support. I've seen clients using this as a shortcut to get the ADL verbs for 1.0.x still. 

Using 'und' in language maps is a particular issue when "canonical" format is requested as the LRS uses language negotiation. If the user's browser doesn't request "*", it could be argued that the LRS is correct not to return "und" even if it's the only language available and most browsers do not request "*" by default. 

Whether or not content negotiation applies to all language maps or just activity definition is ambiguous in 1.0.2 but 1.0.3 is set to clarify that it does apply to all language maps: https://github.com/adlnet/xAPI-Spec/blob/1.0.3/xAPI.md#language-filtering-requirements-for-canonical-format-statements

Today, I don't think any LRS does have a canonical verb display, but the client could still request canonical statements perhaps for the canonical activity definition.

brianjmiller replied:
True, but at the same time we've discussed adding the ability to set the language accept header and it could be argued that there is very little reason outside of a debugging context to have a browser request a statement stream (and even in a debugging context).

Whether it goes away with dropping 0.9 is based on how open we are to backwards compatibility breakage, and with the roadmap to 1.x we are very open to it as a major version bump. The conversion of ids is something I definitely want to see go, if developers are using the short cuts just to get the ADL verbs it should be easy for them to introduce the conversion map in their own code to handle it, never the other OSS libs currently do any of these types of things and developers don't seem to have a problem using a proper verb ID.

brianjmiller commented:
Posted a couple thoughts, we can hash it out further, I know people may not like the 'und' but it was chosen for some decent reasons back when it was originally used. Let me know.

garemoko commented:
Added a response to why I think this is important to tackle. Moved some code inside an if so that display is only calculated from the id if what's passed is in the down convert map (i.e. it's a string like "passed").

garemoko commented:
Closed. May revisit parts of this as part of a 1.0.0 release.
